core/src/main/java/com/datastrato/gravitino/connector/BaseCatalog.java

```diff
@@ -169,12 +184,57 @@ public CatalogOperations ops() {
     return ops;
   }
 
+  public AuthorizationHook getAuthorizationHook() {
+    if (authorization == null) {
+      synchronized (this) {
+        if (authorization == null) {
+          authorization = createAuthorizationInstance();
+        }
+      }
+    }
+    return authorization.hook();
+  }
+
+  private BaseAuthorization<?> createAuthorizationInstance() {
+    String provider = entity.getProperties().get(AUTHORIZATION_IMPL);
+    if (provider == null && provider.isEmpty()) {
+      throw new IllegalArgumentException("Authorization hook provider is not set");
+    }
+
+    ServiceLoader<AuthorizationProvider> loader =
```

Comment:
I'm very puzzled about why we need to use `ServiceLoader` to load the class `AUTHORIZATION_IMPL`, can we directly use class loader?

Reply:
I think SPI is a better way, Anyone can develop the AuthorizationHook plugin, just like Gravitino Catalogs.